src/formatDate.js

```diff
@@ -7,8 +7,62 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparatorFormat = fromFormat[3];
+  const toSeparatorFormat = toFormat[3];
+  const dateObject = {};
+
+  const arrayFromDate = date.split(fromSeparatorFormat);
+
+  fromFormat.forEach((part, i) => {
+    switch (part) {
+      case 'YYYY':
+        dateObject.year = arrayFromDate[i];
+        break;
+
+      case 'YY':
+        const year = parseInt(arrayFromDate[i]);
+
+        dateObject.year =
+          year < 30 ? '20' + arrayFromDate[i] : '19' + arrayFromDate[i];
+        break;
+
+      case 'MM':
+        dateObject.month = arrayFromDate[i];
+        break;
+
+      case 'DD':
+        dateObject.day = arrayFromDate[i];
+        break;
+
+      default:
+        break;
+    }
+  });
+
+  const result = toFormat
+    .map((part) => {
+      if (part === 'YYYY') {
+        return dateObject.year;
+      }
+
+      if (part === 'YY') {
+        return dateObject.year.slice(-2);
+      }
+
+      if (part === 'MM') {
+        return dateObject.month;
+      }
+
+      if (part === 'DD') {
+        return dateObject.day;
+      }
+    })
+    .filter(Boolean)
+    .join(toSeparatorFormat);
```

Comment:
Similar to the `forEach` loop, this `map` operation can be optimized. You can create a mapping object that maps the format parts to their respective handlers, which would help to avoid multiple `if` conditions and make the code more maintainable.